pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -503,6 +503,14 @@ void connectionOpened(final ClientCnx cnx) {
                 cnx.channel().close();
                 return null;
             }
+            
+            if (client.getConfiguration().getConnectionsPerBroker() == 0) {
```

Comment:
uhm... I really don't like to get into the `finalizer()` business... It can be very unpredicably called and you shouldn't do anything that could possibly block..